solr/src/main/java/org/apache/metamodel/solr/SolrDataContext.java

```diff
@@ -0,0 +1,396 @@
+        String queryStr = "*:*";
+
+        if (whereItems != null) {
+            for (int i = 0; i < whereItems.size(); i++) {
+                FilterItem whereItem = whereItems.get(i);
+                queryStr = whereItem.toString();
+            }
+        }
+
+        QueryResponse response = selectRows(table, null, queryStr, 0, 0);
+        return response.getResults().getNumFound();
+    }
+
+    @Override
+    public DataSet executeQuery(Query q) throws MetaModelException {
+        String queryStr = "";
+        QueryResponse response = null;
+
+        Schema schema = getMainSchema();
+        Table table = schema.getTable(0);
+        Column[] columns = table.getColumns();
+
+        boolean selectAll = true;
+        List<SelectItem> selectItems = q.getSelectClause().getItems();
+
+        for (SelectItem selectItem : selectItems) {
+            if (selectItem.getFunction() != null
+                    || selectItem.getColumn() == null) {
+                selectAll = false;
+                break;  
+            }
+        }
+
+        if (selectItems.size() == 1 && selectAll == false) {
+            return super.executeQuery(q);
+        }
+        
+        FilterClause clause = q.getWhereClause();
+
+        if (clause != null) {
+            List<FilterItem> filterItemList = clause.getItems();
+
+            boolean isFirst = true;
+            
+            for (FilterItem filterItem : filterItemList) {
+                String expr = filterItem.getExpression();
+
+                if (expr == null){
+                    expr = filterItem.toString();
+                    expr = expr.replaceAll("[a-z_0-9]+\\.","");
+                }
+                
+                expr = expr.replace(" or ", " OR ").replace(" and ", " AND ").replaceAll("=",":");
+                
+                String joiner = (isFirst == true) ? " " : " AND ";
+                queryStr += joiner + expr;
+                
+                isFirst = false;
+            }
+        }
+        
+        if (queryStr.isEmpty()) {
+            queryStr = "*:*";
+        }
+
+        Integer limitObject = q.getMaxRows();
+        int limit = DEFAULT_LIMIT;
+
+        if (limitObject != null) {
+            limit = (int) limitObject;
+        }
+
+        final List<GroupByItem> groupByItems = q.getGroupByClause().getItems();
+
+        if (groupByItems.size() > 0) {
+            query.clear();
+            query.setQuery(queryStr);
+            query.setFacet(true);
+
+            for (int i = 0; i < groupByItems.size(); i++) {
+                String facetField = groupByItems.get(i).getSelectItem()
+                        .getColumn().getName();
+                query.addFacetField(facetField);
+            }
+
+            setOrder(q, query, QUERYTYPE.FACET);
+
+            query.setFacetLimit(limit);
+
+            try {
+                HttpSolrServer server = initSolrServer();
+                response = server.query(query);
```

Comment:
At this point there has been no regard to the HAVING clause of the query. There might be HAVING items.